taglib/riff/wav/wavproperties.h

```diff
@@ -52,35 +52,107 @@ namespace TagLib {
         /*!
          * Create an instance of WAV::Properties with the data read from the
          * ByteVector \a data.
+         *
+         * \deprecated
          */
         Properties(const ByteVector &data, ReadStyle style);
 
         /*!
          * Create an instance of WAV::Properties with the data read from the
          * ByteVector \a data and the length calculated using \a streamLength.
+         *
+         * \deprecated
          */
         Properties(const ByteVector &data, uint streamLength, ReadStyle style);
 
         /*!
+         * Create an instance of WAV::Properties with the data read from the
+         * ByteVector \a data and the length calculated using \a streamLength
+         * and \a totalSamples.
+         *
+         * \note totalSamples can be zero if the file doesn't have a 'fact' chunk.
+         */
+        Properties(const ByteVector &data, uint streamLength, uint totalSamples, ReadStyle style);
```

Comment:
I have considered another option that we create a new constructor like this:
```
Properties(File *file, ReadStyle style);
```
and move some code from ```WAV::File``` to ```WAV::Properties```. 
It looks straightforward, but it requires to add a ```friend class``` to ```RIFF::File``` and I don't like abusing ```friend```.